packages/server/utils/analytics/analytics.ts

```diff
@@ -109,6 +109,10 @@ export type AnalyticsEvent =
   | 'Snackbar Viewed'
   // Join request
   | 'Join Request Reviewed'
+  // Suggest Groups
+  | 'Suggest Groups Generated'
+  | 'Suggest Groups Clicked'
+  | 'Reset Groups Clicked'
```

Comment:
Should this two metrics be emitted from client side on the actual buttons? I didn't look at the mutations carefully but could there be cases when user clicked the button but the mutation errors out before reaching the metrics line?

Reply:
I thought the preference was to include it on the server because it means less code, and if the client loses their internet connection, the analytics will still be recorded? 

If there's an error in the mutation before we get to the analytics bit, the mutation wouldn't have executed, so would we want to record it? We'd know the error happened as we're recording that in Sentry

Reply:
I still think it's weird to see the metrics emitted from server side when those actions are essentially client side behaviors. Reading through the code though I realize when user clicks the button, inside the mutation it will execute to that line even when the groups were not generated previously so I guess it's essentially the same today? But the code still smells to me. I'll let @Dschoordsch do the arbitration 😄

Reply:
`analytics` is used in server mutations that are triggered by client events in many places throughout the app, such as `createTask`, `startRetrospective`, `addComment`, `addAtlassianAuth`, etc. Do you feel that this is different to those examples?

Reply:
@tianrunhe thanks for tagging me, but I think ultimately you're in control of the analytics and are free to define what you want to track, i.e. if it's more important the user clicked a button vs. the user performed the grouping.
In general I expect server side analytics to be more consistent because of ad-blockers etc. So I'm in favour of the current solution, but it's up to you as I said.
The only odd one I see is `Suggested Groups Generated` because that just happens when advancing the meeting, but I guess it's important to filter the funnel generated->grouping clicked->groups reset